### ISSUE TITLE
Developer signed apps showing as expired

### ISSUE THREAD

**Issue by learntofly83** [sohsatoh/ReProvision-Reborn_old#7]
At the bottom of the screen under “Other Applications” and apps signed with a developer certificate appear.

On matchstick’s version, this would correctly display the value in days until expiry (from expired to 365 days max), however currently even apps that are not revoked and have time left are showing as “expired”

Thanks

**Comment by sohsatoh** [comment]
@learntofly83 Thanks, I will look into it.

**Comment by sohsatoh** [comment]
@learntofly83 

When developing ReProvision Reborn, I have added the ability to detect the deletion of a revision file. It's written in shitty, inefficient code, and also not tested with a developer account.

It does this by searching for the name of the application in all the files in /var/MobileDevice/ProvisioningProfiles, and if it finds it, then it knows that there is a provisioning file for that application.

I don't have a developer account, so please check, but after signing, is there a file in /var/MobileDevice/ProvisioningProfiles 
 where the name of the app is present in the text in the file?

**Comment by learntofly83** [comment]
Hi,

Yes there is a file that relates to the developer signed app in that folder (I also have enterprise apps which are present in there also)

I use a 3rd party signing tool (not Xcode) so these generated files may be slightly different in syntax than Xcode but the apps work (In an unjailbroken state) and previous in reprovision the expiration date was correctly displayed.

WhAt I did notice in the provisioning file was that the app name was not present, however it looks like part of the bundleid with a wildcard WAS present. I’m not a developer (although happy to help debug this) but there are a few other strings that you may wish to look for which appear useful for detecting it’s a developer signed app and perhaps also the UUID may match?


Here’s some interesting strings (with some redaction and replacement with the hash symbol)


```

<key>application-identifier</key>
<string>RLZ84####.*</string>
<key>DeveloperCertificates</key>


<key>TimeToLive</key>
<integer>365</integer>
<key>UUID</key>
<string>6d5288ca-6e7b-40e0-a7dc-############</string> 

```

**Comment by learntofly83** [comment]
Just an update that I’ve noticed, this also affects enterprise signed apps and test flight apps

**Comment by sohsatoh** [comment]
Thanks, this information is really helpful.
I will look into it:)

**Comment by learntofly83** [comment]
No problem and glad to be able to help... not sure if it’s any help to post screenshots but the first is the original reprovision (with reprovisionfix package) and the second is reprovisionreborn with the differences i mentioned :) this is on an iOS 13 device

![500026F3-E261-4DAF-AFBB-D4D96584E257](https://user-images.githubusercontent.com/5455966/111981938-42afa400-8b00-11eb-9a63-14e44fea180b.png)
![2B1A9C08-DCF8-447F-A36F-DE3D1A532A66](https://user-images.githubusercontent.com/5455966/111981941-43e0d100-8b00-11eb-8d8e-5be11aba3895.png)

**Comment by learntofly83** [comment]
No problem and glad to be able to help... not sure if it’s any help to post screenshots but the first is the original reprovision (with reprovisionfix package) and the second is reprovisionreborn with the differences i mentioned :) this is on an iOS 13 device

![E7FEAFC4-46CA-4613-BD7A-7286502B75D2](https://user-images.githubusercontent.com/5455966/111982321-ba7dce80-8b00-11eb-966d-8b65e6fb703d.jpeg)

![8AD91208-F671-4D03-B50C-A79A197FA6AE](https://user-images.githubusercontent.com/5455966/111982313-b81b7480-8b00-11eb-8343-f68b51e42a5e.png)

**Comment by learntofly83** [comment]
No problem and glad to be able to help... not sure if it’s any help to post screenshots but the first is the original reprovision (with reprovisionfix package) and the second is reprovisionreborn with the differences i mentioned :) this is on an iOS 13 device


![B7DD0956-75D2-4B59-8F3E-789282B16E2F](https://user-images.githubusercontent.com/5455966/111982403-d5e8d980-8b00-11eb-9e65-85f73c18ac64.png)

![E37D9DD7-ADC3-4720-B63D-C0FD69CEC1EA](https://user-images.githubusercontent.com/5455966/111982430-ded9ab00-8b00-11eb-9a01-67bef17fbe3e.jpeg)

**Comment by sohsatoh** [comment]
Thank you. This issue has been resolved in the next v0.6.2.
(If you have any problems with v0.6.2 after its release, please open an issue again)
